addons/account_payment/models/payment_transaction.py

```diff
@@ -187,19 +187,18 @@ def _log_message_on_linked_documents(self, message):
         for invoice in self.invoice_ids:
             invoice.message_post(body=message)
 
-    #=== BUSINESS METHODS - GETTERS ===#
+    #=== BUSINESS METHODS - POST-PROCESSING ===#
 
-    def _get_received_message(self):
-        """ Return the message stating that the transaction has been received by the provider.
+    def _finalize_post_processing(self):
+        """ Override of `payment` to write a message in the chatter with the payment and transaction
+        references.
 
-        Note: self.ensure_one()
+        :return: None
         """
-        message = super()._get_received_message()
-
-        if self.state == 'done' and self.payment_id:
```

Comment:
This condition should be re-integrated, should a transaction not have a payment.